app/javascript/profileDropdown/flagButton.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-alert */
+/**
+ * Adds a flag button visible only to trusted users on profile pages.
+ * @function initFlag
+ * @returns {(void|undefined)} This function has no useable return value.
+ */
+export function initFlag() {
+  const flagButton = document.getElementById(
+    'user-profile-dropdownmenu-flag-button',
+  );
+
+  if (!flagButton) {
+    // button not always present when this is called
+    return;
+  }
+
+  /* eslint-disable-next-line no-undef */
```

Comment:
suggestion (non-blocking) - if you add

```
/* global userData */
```

to the top of the file, you can remove the `disable-next-line` here

Reply:
Ah, thanks for this, I had just copied this over from `blockButton.js`, but prefer your solution.